bin/s2nc.c

```diff
@@ -543,8 +543,14 @@ int main(int argc, char *const *argv)
             echo(conn, sockfd);
         }
 
+        /* TODO: Some integration tests fail with this change. This is a work around until we fix those. */
+        /* After the integration tests are fixed, the below shutdown and `if` clause should be replaced with: */
+        /* GUARD_EXIT(s2n_shutdown(conn, &blocked), "Error shutting down connection"); */
         s2n_blocked_status blocked;
-        s2n_shutdown(conn, &blocked);
+        int shutdown_rc = s2n_shutdown(conn, &blocked);
+        if (shutdown_rc == -1 && blocked != S2N_BLOCKED_ON_READ) {
+            return -1;
+        }
```

Comment:
Wouldn't this need to be the opposite to prove that your fix works? Your fix is supposed to stop S2N from blocking on read. How have you verified that your fix works with other implementations?

Reply:
this says that if the call to shutdown fails, then it should only fail because we await a close_notify.

this is unfortunately the case for integration tests.